writer/licence_writer.go

```diff
@@ -18,44 +18,50 @@ func IsLicenceExist(fileName string, licenceFile string) (bool, error) {
 	if err != nil {
 		return false, err
 	}
-	LICENCE := readFile(licenceFile)
+	LICENCE := readFile(licenceFilePath)
 	return strings.Contains(fileContent, LICENCE), err
 }
 
-func IsLicenceFormatValid(fileName string, licenceName string) bool {
-	licence := readFile(licenceName)
-	licenceArray := strings.Split(licence,"\n")
-	firstLineOfLicence := licenceArray[0]
-	fileContent := readFile(fileName)
-	if strings.Contains(fileContent, firstLineOfLicence) {
-		index := strings.Index(fileContent, firstLineOfLicence)
-		tempLicence := string(fileContent[index])
-		for i := index; fileContent[i-1] != '*' || fileContent[i] != '/'; i++ {
-			tempLicence += string(fileContent[i])
+func IsLicenceFormatValid(filePath string, licenceFilePath string) bool {
+	licence := readFile(licenceFilePath)
+	licenceArray := strings.Split(licence, "\n")
+	fileContent := readFile(filePath)
+	for i := 0; i < len(licenceArray); i++ {
+		line := licenceArray[i]
+		if strings.Contains(fileContent, line) {
+			startIndex := strings.Index(fileContent, "/*")
```

Comment:
I think you didn't cover the edge cases for this problem. If you have code like below you would have very unexpected results.
```java
/*
 LICENCE..
*/
public void something(){
   /*
    do something
   */
}
```

Reply:
And by the way the licence finding part can be extracted from the code it would be much more cleaner.